freeipa/src/main/java/com/sequenceiq/freeipa/repository/SyncOperationRepository.java

```diff
@@ -20,10 +20,12 @@
 @AuthorizationResourceType(resource = AuthorizationResource.ENVIRONMENT)
 public interface SyncOperationRepository extends BaseJpaRepository<SyncOperation, Long> {
 
+    // TODO restrict this permission. Set Password and User sync should have more
+    // restrictive permissions than READ but more permissive than WRITE
+    // TODO override other BaseJpaRepository methods for consistent behavior
     @CheckPermission(action = ResourceAction.READ)
     @Override
-    @Query("SELECT s FROM SyncOperation s WHERE s.id = :id")
-    Optional<SyncOperation> findById(@Param("id") Long id);
+    SyncOperation save(SyncOperation entity);
```

Comment:
Why was the findById method removed?

Reply:
it wasn't being used. also, it's already implemented automatically by the base class so there was no reason to override it with a query annotation.